File: OOPS/oopsPython/ConstructorOverloadDemo.py

class ConstructorOverloadDemo:

    def __init__(self):
        self.name = "default"

    # def __init__(self, constructorType):
    #     self.name = constructorType

    def printValues(self):
        print(self.name)

demo1 = ConstructorOverloadDemo()
# demo2 = ConstructorOverloadDemo("parameterized with string")

demo1.printValues()
# demo2.printValues()

File: OOPS/oopsJava/inheritance.polymorphism/ConstructorOverloadDemo.java
package in.kpriet.training.oops;

class ConstructorOverloadDemo {

    String name;
    int number;

    ConstructorOverloadDemo() {
        this.name = "default";
    }

    ConstructorOverloadDemo(String name) {
        this.name = name;
    }

    ConstructorOverloadDemo(int number) {
        this.number = number;
    }

    void printString() {
        System.out.println(this.name);
    }

    void printInt() {
        System.out.println(this.number);
    }

    public static void main(String[] args) {
        
        ConstructorOverloadDemo demo1 = new ConstructorOverloadDemo();
        ConstructorOverloadDemo demo2 = new ConstructorOverloadDemo("parameterized");
        ConstructorOverloadDemo demo3 = new ConstructorOverloadDemo(1);

        demo1.printString();
        demo2.printString();
        demo3.printInt();
    }
}